### ISSUE TITLE
feat: templates

### ISSUE THREAD

**Issue by PencilMing** [jacobtread/obsidian-timekeep#5]
Hello jacobtread, I really like your plugin, but may I ask if Timekeep can be set up with templates? I tried to set up the steps commonly used in my project as a template, but I failed. For example, in my video projects, there are often four steps: script, copywriting, shooting, and post-production. I hope that when the project template is established, Timekeep will have these four items directly, and then I only need to click on the corrsponding item to start recording time. Now I can only enter manually and then start.

**Comment by jacobtread** [comment]
Hey, I'll check this out tonight (NZST).

Here’s what I’m thinking for how to implement this:

- Add a new section in settings to create "templates," which will just have a name.
- You can add new templates.
- The template names will show up in the usual start area with a play button next to them.
- Clicking the play button will start a new section using the template name.

Is this what you were looking for?

**Comment by PencilMing** [comment]
Thank you for your reply. I was thinking if it might be simpler this way: for example, I could integrate with the templater plugin and edit the code in the template as shown in Figure 1. 
Figure 1 
![CleanShot 2024-06-23 at 21 16 10@2x](https://github.com/jacobtread/obsidian-timekeep/assets/82462283/77757af7-9dbb-4838-b295-cc8d6b9c6b19)
This would allow the time view function to work properly, as shown in Figure 2, and it would also allow me to enter the start time by clicking 'start' normally.
Figure 2
![CleanShot 2024-06-23 at 21 16 27@2x](https://github.com/jacobtread/obsidian-timekeep/assets/82462283/0f060edf-a59d-4cb3-909e-283993d1a3d9)
But in this vision i get Figure3, it's a wrong view.
Figure3
![CleanShot 2024-06-23 at 21 22 13@2x](https://github.com/jacobtread/obsidian-timekeep/assets/82462283/c0deba9d-4742-4108-92d9-40ff77f1fe72)
That's roughly what I mean. Of course, I'm not sure if the feature you suggested would be more work to implement than the method I mentioned.

**Comment by jacobtread** [comment]
Currently, there is no concept of an entry that hasn't been started yet, so trying to press play on those entries would just create another entry and convert that entry to a group 

I think I'll have to modify the underlying data model to allow the start time to be null to represent entries that haven't started (then you'd use a null start and end time in your template)

And I'd add an additional check in the start logic for entries with a null start time to start the entry rather than creating a new one

Will give this a look likely tomorrow night and see what I can do

**Comment by jacobtread** [comment]
@PencilMing I have implemented this in #6 and it should now be released as v1.3.0 of the plugin. Rather than empty strings for the startTime and endTime this uses null (As empty strings will attempt to be parsed as dates and fail)

An example:
````
```timekeep
{"entries":[{"name":"Block 1","startTime":null,"endTime":null,"subEntries":null}]}
```
````

**Comment by PencilMing** [comment]
Thank you very much! I will make a video for this plugin and further experience the plugin.
